### ISSUE TITLE
Autotuner invalid value for tiling

### ISSUE THREAD

**Issue by Artix18** [facebookresearch/TensorComprehensions#562]
Built from source (master), commit b77b5835ca1165f63434e09ca915eea81df07726

input sizes: N, C, H, W, O, kH, kW = 32, 4, 56, 56, 16, 1, 1
Therefore, if i am not mistaken, the union of "divisors" (and {0}) should be:
[1, 2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 14, 16, 19, 28, 56, 32, 64, 128, 256, 0]

```python
import tensor_comprehensions as tc
import torch

tc_name = "convolution"
tc_code = """
  def convolution(float(N,C,H,W) I, float(M,C,KH,KW) W1) -> (O) {
  O(n, m, h, w) +=! I(n, r_c, h + r_kh, w + r_kw) * W1(m, r_c, r_kh, r_kw)
  }
"""

N, C, H, W, O, kH, kW = 32, 4, 56, 56, 16, 1, 1
I, W1 = torch.randn(N, C, H, W, device='cuda'), torch.randn(O, C, kH, kW, device='cuda')
inp = (I, W1)

opts = tc.MappingOptions("naive")
opts.tile([5])

tc.autotune(tc_code, tc_name, *inp, starting_options=opts)
```

```
Traceback (most recent call last):
  File "bug.py", line 18, in <module>
    tc.autotune(tc_code, tc_name, *inp, starting_options=opts)
  File "/private/home/artix18/anaconda3/envs/tc_build/lib/python3.6/site-packages/tensor_comprehensions/__init__.py", line 214, in autotune
    tuner_config)
ValueError: RangeParameter::selectFromValue: value 5 not in range. The valid values are: 1 2 4 7 8 14 16 28 32 56 64 128 256 0
```

In my opinion, 5=ceil(32/7) and should then be a valid value.
I didn't have this problem previously.

**Comment by nicolasvasilache** [comment]
Hmm, see [here](https://github.com/facebookresearch/TensorComprehensions/blob/master/tc/autotuner/utils.cc#L44).
We actually set the ceil by powers2 only, you can use that instead of all ceils.
But the impl seems bugged (cc @ttheodor) because we insert in place and reuse iteratively.

@Artix18 this is a good opportunity to commit an easy code fix to C++-land :)
Keeping 'ceil by powers2 only' behavior sounds reasonable to me to cut the size of the search space.

**Comment by ftynse** [comment]
The function @nicolasvasilache pointed out divides sizes _by_ powers of two, not vice versa.  So, it will include ceil(7/32)=1 but not ceil(32/7)=5.  The idea was, I think, that each time has a power-of-two number of iterations.

The actual bug here is the autotuner failing because the starting options passed by the user do not correspond to those the tuner can select.  I'd say the user knows better and the tuner should include those options in the pool of allowed values.

**Comment by nicolasvasilache** [comment]
@ftynse  it's `ceil(56/8) = 7` such that the result of `56/7` is a power of 2. 
In Jules' case, there is no power of 2 for the problem sizes such that `ceil(size / 2^k) == 5`.
It makes sense to say the user knows best and to allow it.

**Comment by ftynse** [comment]
> @ftynse it's ceil(56/8) = 7 such that the result of 56/7 is a power of 2.

I don't get what are you trying to say there.  Even if `56/8=7`, there is no reason why `7` would appear on the RHS of the division to get you `ceil(32/7)`.

We agree on the path to follow here, though.

**Comment by ftynse** [comment]
closed by #564
